analyze.py

```diff
@@ -224,110 +236,131 @@ def parse_requirements_txt(self, changes):
             if name_ver_match := re.match(name_ver_pat, changes[cur]):
                 name = name_ver_match.groups()[0]
                 ver = name_ver_match.groups()[1]
-                pkg_ver.append((name,ver))
+                pkg_ver.append((name, ver))
             cur += 1
 
         print(f"[DEBUG]: pkg_ver length: {len(pkg_ver)}")
         return pkg_ver
 
+    def parse_poetry_lock(self, changes):
+        """Parse lines added to a poetry.lock file to identify package names and versions."""
+        file_name_pat = re.compile(
+            r"""^       # match beginning of string
+            \s{4}       # start with four spaces
+            {file       # start of the mapping for a file
+            \s=\s       # whitespace separated mapping assignment operator
+            "(.*?)"     # non-greedy capture group for the file name
+            """,
+            re.VERBOSE,
+        )
```

Comment:
The `poetry.lock` file format is not a documented standard. It is TOML. The regex in this parsing function is looking for file names in the `metadata.files` table...which is updated to include all the files that _could_ be installed to satisfy the individual `package` table entry for each dependency. Using this section is ideal here b/c the patch/diff will show the line as added for every dependency change and the package name/version tuple is available on that single line.

The filename will need to be either a source distribution (aka sdist) or a wheel in order to be recognized. There may be some odd edge cases...particularly for _very_ old packages...but this assumption seems to hold for the vast majority of cases.